include/extractor/guidance/sliproad_handler.hpp

```diff
@@ -42,6 +47,50 @@ class SliproadHandler : public IntersectionHandler
     Intersection operator()(const NodeID nid,
                             const EdgeID via_eid,
                             Intersection intersection) const override final;
+
+  private:
+    util::ScopedGeojsonLoggerGuard<util::NodeIdVectorToLineString> geojson_lines;
+    util::ScopedGeojsonLoggerGuard<util::NodeIdVectorToMultiPoint> geojson_points;
+
+    struct IntersectionAndNode final
+    {
+        Intersection intersection;
+        NodeID node;
+    };
+
+    // Returns a potential next intersection along a road skipping over traffic lights.
+    boost::optional<IntersectionAndNode> getNextIntersection(const NodeID at,
```

Comment:
should probably not be a part of the sliproad handler but at least of the base class, allowing all handlers to find next intersections